core/node/base_token_fetcher/src/lib.rs

```diff
@@ -105,12 +112,14 @@ impl BaseTokenFetcher {
             }
         }
 
-        let conversion_rate = conversion_rate
+        let conversion_rate = conversion_rate_str
             .ok_or_else(|| anyhow::anyhow!("Failed to fetch the native token conversion rate"))?;
+        let conversion_rate = BigDecimal::from_str(&conversion_rate)
```

Comment:
Following my comment above, this from_str returns a result, right?
So from_str already reports what was trying https://docs.rs/bigdecimal/latest/bigdecimal/enum.ParseBigDecimalError.html
to parse, so ok then, ignore what I said above.